plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -215,19 +219,51 @@ def _handle_group_category(type_id, unit_key, metadata, file_path, conduit, conf
     :type  conduit: pulp.plugins.conduits.upload.UploadConduit
     :type  config: pulp.plugins.config.PluginCallConfiguration
     """
+    # check whether a file was uploaded, and treat this a upload of comps.xml file
+    if file_path is not None and os.path.getsize(file_path) > 0:
+        repo_id = repo.id
```

Comment:
I would just use repo.id everywhere and drop this line, but that's personal preference I suppose.